### ISSUE TITLE
Turn off compiler warning by warning id

### ISSUE THREAD

**Issue by ylazy** [airsdk/Adobe-Runtime-Support#2313]
Sometimes I need to turn off the compiler warning by id, for example this code:
```as
public static function isDescendant(objectOrClass:*, isDescendantOfClass:Class):Boolean
{
	if (objectOrClass == null || isDescendantOfClass == null) return false;
	
	var thisClass:Class = objectOrClass is Class ? objectOrClass : getClass(objectOrClass);
	
	// see https://www.jacksondunstan.com/articles/1440
	return (thisClass == isDescendantOfClass || thisClass.prototype instanceof isDescendantOfClass);
}
```
will show up the compiler warning: `The instanceof operator is deprecated, use the is operator instead. (3555)`
It would be nice if I can turn it off by something like `-compiler.hide-actionscript-warnings=3555,3596,3597`

Thanks!

**Comment by ajwfrost** [comment]
There's a setting to turn all of them off:
`-compiler.show-actionscript-warnings=false`
but I can see the need for a setting like you've suggested.

But when I compile that function, I don't see the error number like you have in brackets after it.. are you using the Flex compiler for this? Actually I don't even see that with the Flex compiler..?

**Comment by ylazy** [comment]
>but I can see the need for a setting like you've suggested.

Yes, I don't want to turn all warnings off. Some of them are useful for me.

>I don't see the error number like you have in brackets after it

I think it's an as3mxml plugin feature. The Output tab doesn't show, but the Problem tab does.

![image](https://user-images.githubusercontent.com/6010183/203269821-cc62f61e-310f-4ce6-9068-633eb5cc0c75.png)

![image](https://user-images.githubusercontent.com/6010183/203270363-8d06c383-69c7-413c-9435-490212386af0.png)

**Comment by ajwfrost** [comment]
Ah - interesting! So that sounds like Josh being very clever with the code analysis stuff! The Java classes that report the errors do have these codes in them so maybe we need to output that too.

Although I was just looking at the documentation:
https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/compilerWarnings.html
1084 is there but 1554 isn't? 3555 is... 

Anyway, this should be pretty straightforward to add :-)

**Comment by ylazy** [comment]
>1554 isn't?

I found out that 1554 is [a royale compiler problem id](https://github.com/apache/royale-compiler/blob/fa0166cb70591196c87bc5ac350bbbad7e075467/compiler/src/main/java/org/apache/royale/compiler/problems/ConflictingDefinitionProblem.java#L34).
